tools/check-metadata.py

```diff
@@ -0,0 +1,155 @@
+import os
+import subprocess
+import sys
+from collections import OrderedDict
+
+
+ROOT_DIR = os.path.abspath(
+    os.path.join(os.path.dirname(os.path.realpath(__file__)), os.pardir))
+TOOLS_DIR = os.path.join(ROOT_DIR, "tools")
+DATA_DIR = os.path.join(ROOT_DIR, "data")
+SOUNDS_DIR = os.path.join(DATA_DIR, "sounds")
+
+SYSTEM_METADATA_PATH = os.path.join(DATA_DIR, "metadata.json")
+METADATA_SCHEMA_PATH = os.path.join(ROOT_DIR, "ci", "metadata.schema.json")
+
+
+def check_media_file_integrity(filepath):
+    stream = ffmpeg.input(filepath).output("pipe:", format="null")
+    try:
+        stream.run(capture_stdout=True, capture_stderr=True)
+    except ffmpeg.Error:
+        return False
+    return True
+
+
+def check_sorted(metadata):
+    if not sorted(metadata) == list(metadata):
+        return False
+
+    for sound in metadata:
+        props = list(metadata[sound])
+        if not sorted(props[1:]) == props[1:]:
+            return False
+
+
+def check_files(metadata, check_media_integrity=False):
+    valid = True
+    for sound in metadata:
+        location = metadata[sound].get("sound-file")
+        if not location:
+            locations = metadata[sound].get("sound-files")
+        else:
+            locations = [location]
+
+        for location in locations:
+            full_location = os.path.abspath(os.path.join(SOUNDS_DIR, location))
+            file_extension = os.path.splitext(full_location)[1]
+
+            if not os.path.isfile(full_location):
+                valid = False
+
+                print(f"File '{full_location}' does not exist.",
+                      file=sys.stderr)
+            elif (check_media_integrity and
+                    not check_media_file_integrity(full_location)):
+                valid = False
+                print(f"File '{full_location}' is not a valid media file.",
+                      file=sys.stderr)
+            if file_extension != ".webm":
+                print(f"File '{full_location}' is not a 'webm' file. "
+                      "Did you forget to convert it? "
+                      "Use ./tools/convert-sounds.sh", file=sys.stderr)
+    return valid
+
+
+def check_schema():
+    proccess = subprocess.Popen(
+        ["jsonschema", "-i", f"{args.path.name}", f"{METADATA_SCHEMA_PATH}"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE
+    )
+    out, err = proccess.communicate()
+    is_valid = proccess.returncode == 0
+
+    if not is_valid:
+        print(err.decode("utf-8"), file=sys.stderr)
+    return is_valid
+
+
+def check_json():
+    proccess = subprocess.Popen(
+        ["python", "-m", "json.tool", f"{args.path.name}"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE
+    )
+    out, err = proccess.communicate()
+    is_valid = proccess.returncode == 0
+
+    if not is_valid:
+        print(err.decode("utf-8"), file=sys.stderr)
+    return is_valid
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
```

Comment:
I would generally try to avoid "Swiss army scripts" that do multiple things that are all optional and can be specified in any combination. Either we should do all the checks all the time, or have a very good reason to skip some of the checks.

Reply:
I was thinking this script could be modified in the future to be used by David or game designers/developers to check the integrity of their local metadata files, too. In this case, they wouldn't be care probably if the metadata is sorted.

So, I would say to remove `--check-files` and make it the default behavior always. I would left `--check-integrity` optional because you need the right decoders and the ffmpeg-python library to check that.

Let me know what you think, please.

Reply:
I would say we're not concerned with that right now. It's unlikely David will be comfortable using any more command line tools than strictly necessary.

If we do decide to do that, then I'd say we should make a separate shell script which follows the workflow that David wants. We should stick to whatever we need for CI, and make all of those checks mandatory.